src/nanoemoji/write_font.py

```diff
@@ -229,51 +236,114 @@ def _next_name(ufo: ufoLib2.Font, name_fn) -> str:
     return name_fn(i)
 
 
-def _create_glyph(color_glyph: ColorGlyph, painted_layer: PaintedLayer) -> Glyph:
+class ReuseResult(NamedTuple):
+    glyph_name: str
+    transform: Affine2D
+
+
+class GlyphReuseCache:
+    def __init__(self, config: FontConfig):
+        self._config = config
+        self._known_glyphs = set()
+        self._reusable_paths = {}
+
+    def try_reuse(self, path: str) -> Optional[ReuseResult]:
+        """Try to reproduce path as the transformation of another glyph.
+
+        Path is expected to be in font units.
+
+        Returns (glyph name, transform) if possible, (input path, identity) if not.
+        """
+        assert (
+            not path in self._known_glyphs
+        ), f"{path} isn't a path, it's a glyph name we've seen before"
+        print("try_reuse", path)
+        assert path.startswith("M"), f"{path} doesn't look like a path"
+
+        if self._config.reuse_tolerance == -1:
+            print("  reuse disabled")
+            return None
+
+        norm_path = normalize(SVGPath(d=path), self._config.reuse_tolerance).d
+        if norm_path not in self._reusable_paths:
+            print(f"  not in reusable {norm_path}")
+            return None
+
+        glyph_name, glyph_path = self._reusable_paths[norm_path]
+        affine = affine_between(
+            SVGPath(d=glyph_path), SVGPath(d=path), self._config.reuse_tolerance
+        )
+        if affine is None:
+            print(f"  affine_between failed {norm_path}")
```

Comment:
it used to be a logging.warning when we grouped a path but no `affine_between` could be found (signalling a potential picosvg bug), maybe we keep that?

BTW, all print() must go ;)